src/lib.rs

```diff
@@ -90,282 +85,142 @@ macro_rules! implement {
 implement!(LowerHex, "{:02x}");
 implement!(UpperHex, "{:02X}");
 
-/// A context.
-#[derive(Clone)]
-pub struct Context {
-    buffer: [u8; 64],
-    count: [u32; 2],
-    state: [u32; 4],
-}
+// Constants
+const PADDING: [u8; 64] = {
+    let mut padding = [0; 64];
+    padding[0] = 0x80;
+    padding
+};
 
-const PADDING: [u8; 64] = [
-    0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
-    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+#[rustfmt::skip]
+const SHIFTS: [u32; 64] = [
+    07, 12, 17, 22, 07, 12, 17, 22, 07, 12, 17, 22, 07, 12, 17, 22, // Round 1
+    05, 09, 14, 20, 05, 09, 14, 20, 05, 09, 14, 20, 05, 09, 14, 20, // Round 2
+    04, 11, 16, 23, 04, 11, 16, 23, 04, 11, 16, 23, 04, 11, 16, 23, // Round 3
+    06, 10, 15, 21, 06, 10, 15, 21, 06, 10, 15, 21, 06, 10, 15, 21, // Round 4
 ];
 
-impl Context {
-    /// Create a context for computing a digest.
-    #[inline]
-    pub fn new() -> Context {
-        Context {
-            buffer: [0; 64],
-            count: [0, 0],
-            state: [0x67452301, 0xefcdab89, 0x98badcfe, 0x10325476],
-        }
-    }
+// f64::floor(power * f64::abs(f64::sin(i as f64 + 1.0))) as u32
+const SIN_CONSTS: [u32; 64] = [
```

Comment:
Please call it `SINES` to match `SHIFTS`.

Reply:
Done